qcodes/instrument/visa.py

```diff
@@ -24,37 +24,36 @@ class VisaInstrument(Instrument):
     Base class for all instruments using visa connections.
 
     Args:
-        name (str): What this instrument is called locally.
-
-        address (str): The visa resource name to use to connect.
-            Optionally includes '@<backend>' at the end. For example,
-            'ASRL2' will open COM2 with the default NI backend, but
-            'ASRL2@py' will open COM2 using pyvisa-py. Note that qcodes
-            does not install (or even require) ANY backends, it is up to
-            the user to do that. see eg:
-            http://pyvisa.readthedocs.org/en/stable/names.html
-
-        timeout (int, float): seconds to allow for responses. Default 5.
-
+        name: What this instrument is called locally.
+        address: The visa resource name to use to connect.
+        timeout: seconds to allow for responses. Default 5.
         terminator: Read termination character(s) to look for. Default ``''``.
-
         device_clear: Perform a device clear. Default True.
-
+        visalib: Visa backend to use when connecting to this instrument.
+            This should be in the form of a string '@<backend>'.
+            By default the NI backend is used, but '@py' will use the
+            ``pyvisa-py`` backend. Note that QCoDeS does not install (or even require)
+            ANY backends, it is up to the user to do that. see eg:
+            http://pyvisa.readthedocs.org/en/stable/names.html
         metadata (Optional[Dict]): additional static metadata to add to this
             instrument's JSON snapshot.
 
-    See help for ``qcodes.Instrument`` for additional information on writing
+    See help for :class:`qcodes.Instrument` for additional information on writing
     instrument subclasses.
 
     Attributes:
         visa_handle (pyvisa.resources.Resource): The communication channel.
     """
 
-    def __init__(self, name, address=None, timeout=5,
-                 terminator='', device_clear=True, visalib=None, **kwargs):
+    def __init__(self, name: str, address: str, timeout: Union[int, float] = 5,
```

Comment:
I do not see the point of the default value of address being None. This however does break a test of simulated instruments

Reply:
that's good, i agree.